packages/@lwc/ssr-compiler/src/compile-template/transformers/element.ts

```diff
@@ -66,13 +62,26 @@ const bYieldDynamicValue = esTemplateWithYield`
 
         if (attrValue !== undefined && attrValue !== null) {
             yield ' ' + attrName;
+
             if (attrValue !== '') {
-                yield \`="\${prefix}\${htmlEscape(String(attrValue), true)}"\`;
+                yield \`="\${htmlEscape(String(attrValue), true)}"\`;
             }
         }
     }
 `<EsBlockStatement>;
 
+const bYieldClassDynamicValue = esTemplateWithYield`
+    {
+        const attrValue = normalizeClass(${/* attribute value expression */ is.expression});
+        const shouldRenderScopeToken = hasScopedStylesheets || hasScopedStaticStylesheets(Cmp);
+
+        if (attrValue) {
+            const prefix = shouldRenderScopeToken ? stylesheetScopeToken + ' ' : '';
+            yield \` class="\${prefix}\${htmlEscape(String(attrValue), true)}"\`;
+        }
+    }
+`<EsBlockStatement>;
+
```

Comment:
I think it would be cleaner to put this inline in `bYieldDynamicValue`. We already have the attr name, so we can just add `if (attrName === 'class') { /* do something */ }`. Rollup/minifiers will strip unused code.

Reply:
Ah wait I see why you did this. The `class` code is sufficiently different from the non-`class` code (e.g. it needs the scope token). OK this is fine then.

Reply:
Yeah, and it will get even more different to fix the expected failures. Although I think it will take some runtime code to fix completely.

I was able to get the diff even closer, but the changes would be outside the scope of this PR.

This separation helped a lot with debugging.